pkg/kubelet/gpu/nvidia/util/util.go

```diff
@@ -0,0 +1,37 @@
+/*
+Licensed under the Apache License, Version 2.0 (the "License");
+you may not use this file except in compliance with the License.
+You may obtain a copy of the License at
+
+    http://www.apache.org/licenses/LICENSE-2.0
+
+Unless required by applicable law or agreed to in writing, software
+distributed under the License is distributed on an "AS IS" BASIS,
+WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+See the License for the specific language governing permissions and
+limitations under the License.
+*/
+
+package util
+
+import (
+	"os"
+	"path/filepath"
+	"regexp"
+)
+
+// Get all the NVIDIA GPU cards' path from /dev/
+func Discovery() ([]string, error) {
+	gpulist := []string{}
+	err := filepath.Walk("/dev", func(path string, f os.FileInfo, err error) error {
+		reg := regexp.MustCompile(`^nvidia[0-9]*$`)
+		gpupath := reg.FindAllString(f.Name(), -1)
+		if gpupath != nil && gpupath[0] != "" {
+			gpulist = append(gpulist, "/dev/"+gpupath[0])
+		}
+
+		return nil
+	})
+
```

Comment:
Ok, I would like to add another method about the /dev/nvidiactl and /dev/nvidia-uvm check and use it in the Nvidia Init function, if they don't exist, the NVIDIA plugin should be nil.